UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs

```diff
@@ -854,6 +930,38 @@ public void RestartHeart()
 			}
 		}
 
+		[Server]
+		public void ApplyReagentsToSurface(ReagentMix Chemicals, BodyPartType bodyPartAim)
+		{
+
+			foreach (var reaction in ALLExternalMetabolismReactions)
+			{
+				if (reaction.HasInitialTouchCharacteristics)
+				{
+					if (reaction.HasIngredients(Chemicals))
+					{
+						var Amount =  reaction.GetReactionAmount(Chemicals);
+						foreach (var TouchCharacteristics in  reaction.InitialTouchCharacteristics)
+						{
+							ApplyDamageToBodyPart(this.gameObject, Amount * TouchCharacteristics.EffectPerOne,
+								TouchCharacteristics.AttackType,
+								TouchCharacteristics.DamageEffect, bodyPartAim);
+						}
+					}
+				}
+			}
+
+			SurfaceReagents[bodyPartAim].Add(Chemicals);
+
+			if (SurfaceReagents[bodyPartAim].Total  > 5)
```

Comment:
I think the `5` here should be a class variable or constant, as you're using it twice (and maybe it should be configurable).